src/nickel.rs

```diff
@@ -156,7 +157,15 @@ impl<D: Sync + Send + 'static> Nickel<D> {
         });
 
         let server = Server::new(self.middleware_stack, self.data);
-        let listener = server.serve(addr, self.keep_alive_timeout).unwrap();
+
+        let is_test_harness = env::vars().any(|(ref k, _)| k == "NICKEL_TEST_HARNESS");
```

Comment:
The server in the example will fail to bind to the port "6767" as it's in use, which will currently panic.

Builds are sequential, but testing (and running the build artifacts) is still in parallel. Sadly at least on my machine, likely due to multirust, a call to `cargo build` when it builds nothing still takes a second, so it often means things happen sequentially anyway, but occasionally I do see some tests happening in parallel. This will definitely be true for more complicated tests which may be less instantaneous.